File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
INFINITY = math.inf


class Solution:
    def uniformArray(self, nums: list[int]) -> bool:
        mn = INFINITY

        for num in nums:
            if num % 2 != 0:
                mn = min(mn, num)

        for num in nums:
            if num % 2 == 0 and mn != INFINITY and mn > num:
                return False

        return True

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.js
/**
 * @param {number[]} nums1
 * @return {boolean}
 */
var uniformArray = function (nums1) {
    let min = Infinity;

    for (let num of nums1) {
        if (num % 2 !== 0) {
            min = Math.min(min, num);
        }
    }

    for (let num of nums1) {
        if (num % 2 === 0 && min !== Infinity && min > num) {
            return false;
        }
    }

    return true;
};